dfget/core/downloader/p2p_downloader/p2p_downloader.go

```diff
@@ -241,7 +241,7 @@ func (p2p *P2PDownloader) pullPieceTask(item *Piece) (
 		if p2p.queue.Len() > 0 {
 			break
 		}
-		sleepTime := time.Duration(rand.Intn(1400)+600) * time.Millisecond
+		sleepTime := netutils.CalculateTimeout(int64(item.PieceSize), p2p.cfg.MinRate, config.DefaultMinRate)
```

Comment:
The sleeping here is to wait for supernode to prepare an available piece. But the `MinRate` means the minimum speed of `dfget` downloading a piece. We cannot use the `MinRate` to calculate the `sleepTime`.